base/error.go

```diff
@@ -38,6 +39,22 @@ func ErrorAsHTTPStatus(err error) (int, string) {
 	if err == nil {
 		return 200, "OK"
 	}
+
+	switch err.Error() {
+	case gocbcore.ErrKeyNotFound.Error():
+		return http.StatusNotFound, "missing"
+	case gocbcore.ErrKeyExists.Error():
+		return http.StatusConflict, "Conflict"
+	case gocbcore.ErrTimeout.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	case gocbcore.ErrOverload.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	case gocbcore.ErrBusy.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	case gocbcore.ErrTmpFail.Error():
+		return http.StatusServiceUnavailable, "Database server is over capacity"
+	}
+
```

Comment:
Generally I'm disappointed that we still end up having to do a string compare for gocb errors, instead of being able to include them in the type switch below.  Feels like there's the potential for confusion when one of the error types below happens to resolve to the same string as one of these gocbcore errors.